src/splice.ts

```diff
@@ -1,12 +1,104 @@
+import type {
+  ArraySplice,
+  GreaterThan,
+  IsNegative,
+  Subtract,
+  Sum,
+} from "type-fest";
+import type {
+  CoercedArray,
+  IterableContainer,
+  TupleParts,
+} from "./internal/types";
 import { purry } from "./purry";
 
+type FixedLengthSplice<
```

Comment:
I'd simplify this with more helper utility types. The logic is pretty straightforward, but it isn't apparent when inlined like this.

This is how I simplified it while reading the code:

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgSQM4DkCmBzAhjYAN0wBo4BlAVxDIHEpN9MoAVAC1wDsyqAjGKLgDGMMgEEoggJ7kwAG2BDMcAL5wAZlAgg4AIhhSwmALTrMqGLoDcAKBsGjKGM1y85mAMIROMXME7MADwscAC8cJScANacEADunAB8YXAMuAAm3nJScCwA2gC6cAA+cIW29obKyOoAMsDOgnKB6HCYAB7OnOmocJzUvMxkAGK4cnK8wlHJ4f0gg1BtnZjdva0A-HCj45NCUXAAXHDoFQ7KYiKUYwDy6uqomDCBNnC5S109Ti5unt6+-kMXsd3itPnMFiQbDMUBgcPgiJgWskOh9euoxg8gZt0ECjuDmCDVrk8rp3JxsDA2LoCli+gNmLiUHUGi5mlQQMESWSKVSCmR0IkyAAGRKnKpwC4wK5yWorHnPV4hFGg3rIRqudxeHx+AJQSGvW73R6EsH0vVA1rKon483QtBYPAEYhIk29ASUTC0oWMm2u4mkuWU6m0m2M+iMRrsLiBdmBQ0PUTHQX+7lBgrI5ZE92e16bGr1dVsyj8QQiTkB8lpsjxx7JkWMk52M5bYDtTDpWWVtiyBRKBVvK2fNXfTV-HWA17kXywP02-VwAAimHczi8kXgg96c6BACVMPJhJgQCsN5mh+qflr-rr5wBRO6YEQIqe4GebunzAnhSXSmtPFg8NOMCClCKQSNIPaKIiQIAUC95mE+xAvrA84-mMnbygBcDwY+TqYMhiZLiuvzriBrx7geSjHj4UK2EAA